tests/foreman/ui/test_ansible.py

```diff
@@ -565,6 +565,54 @@ def test_positive_ansible_config_report_changes_notice_and_failed_tasks_errors(
             assert f'notice Install the {package_name} package' in notice_log['permission_denied']
             assert f'Installed: rubygem-{package_name}' in notice_log['permission_denied']
 
+    def test_positive_non_admin_hostgroup_permission(
+        self, request, function_org, function_location, target_sat
+    ):
+        """Verify Hostgroup page is accessible to non-admin user.
+
+        :id: 3b8ce895-9d4c-491d-a5eb-d292748134a5
+
+        :customerscenario: true
+
+        :BZ: 2236418
+
+        :steps:
+            1. Create a user with non-admin permissions and assign "Ansible Roles Manager" and "Ansible Tower Inventory Reader"
+            to the user.
+            2. Create a hostgroup and add in same organization and location as user.
+            3. Login to Satellite WebUI with user credentials and navigate to Configure -> Hostgroups
+
+        :expectedresults: The hostgroup created should be visible to non-admin user.
+        """
+        password = settings.server.admin_password
+
+        ansible_manager_role = (
+            target_sat.api.Role().search(query={'search': 'name="Ansible Roles Manager"'})[0].id
+        )
+        tower_inventory_manager_role = (
+            target_sat.api.Role()
+            .search(query={'search': 'name="Ansible Tower Inventory Reader"'})[0]
+            .id
+        )
```

Comment:
```suggestion
        roles = ['Ansible Roles Manager', 'Ansible Tower Inventory Reader']
        user_role = [target_sat.api.Role().search(query={'search': f'name="{role}"'})[0].id for role in roles]
        SELECTED_ROLE = [user_role[0], user_role[1]]
```

Reply:
You can update the code this way if you prefer.

Reply:
It would take the same amount of time so I would rather go with the existing implemetation.